command/loadbot/helper.go

```diff
@@ -72,58 +70,94 @@ func calculateAvgBlockUtil(gasData map[uint64]GasMetrics) float64 {
 	return sum / float64(len(gasData))
 }
 
-// fetch block gas usage and gas limit and calculate block utilization
-func (l *Loadbot) calculateGasMetrics(jsonClient *jsonrpc.Client, gasMetrics *BlockGasMetrics) error {
-	errGr, _ := errgroup.WithContext(context.Background())
+// getBlockGasMetrics fetches block gas metrics from the JSON-RPC client
+// for the specified blocks
+func getBlockGasMetrics(
+	jsonClient *jsonrpc.Client,
+	blockNums map[uint64]struct{},
+) (*BlockGasMetrics, error) {
+	var (
+		errors          = make([]error, 0)
+		blockGasMetrics = &BlockGasMetrics{
+			Blocks: make(map[uint64]GasMetrics),
+		}
+		errorsLock sync.Mutex
```

Comment:
Why not swap the order between `blockGasMetrics` and `errorsLock`?

Reply:
You're right, it makes more sense to bundle them 👍 

Fixed:

[09a7e48](https://github.com/0xPolygon/polygon-edge/pull/497/commits/09a7e489dd31e869163596a27abb7e13c9849b10)